src/Model/Group.php

```diff
@@ -300,20 +317,51 @@ public static function removeMemberByName($uid, $name, $cid)
 	/**
 	 * @brief Returns the combined list of contact ids from a group id list
 	 *
+	 * @param int     $uid
 	 * @param array   $group_ids
 	 * @param boolean $check_dead
 	 * @return array
 	 * @throws \Exception
 	 */
-	public static function expand($group_ids, $check_dead = false)
+	public static function expand($uid, array $group_ids, $check_dead = false)
 	{
 		if (!is_array($group_ids) || !count($group_ids)) {
 			return [];
 		}
 
-		$stmt = DBA::select('group_member', ['contact-id'], ['gid' => $group_ids]);
-
 		$return = [];
+
+		$key = array_search(self::FOLLOWERS, $group_ids);
+		if ($key !== false) {
+			$followersStmt = Contact::select(['id'], [
+				'uid' => $uid,
+				'rel' => [Contact::FOLLOWER, Contact::FRIEND],
+				'protocol' => Protocol::SUPPORT_PRIVATE,
+			]);
+
+			while($follower = DBA::fetch($followersStmt)) {
```

Comment:
Please add a space behind the "while". Also I don't guess that this will work as expected. As I just realized, the ```Contact::select``` isn't doing what it should be. It is calling ```DBA::toArray``` so it already returned an array.

This is a serious design difference to how the identical named ```DBA``` and ```Item``` functions are working. They do return a database handle that has to fetched via the appropriate ```::fetch``` functions.

I strongly vote for changing the behaviour of that function accordingly or to rename the function into something like ```Contact::selectToArray```.

BTW: I guess I will add some small little PR right to for a "DBA::selectToArray" function.